src/AssetsPage/AssetsTable/index.jsx

```diff
@@ -116,6 +139,9 @@ export class AssetsTable extends React.Component {
       // TODO: create lockButton
       currentAsset.delete_asset = deleteButton;
       // TODO: currentAsset.lock_asset = lockButton;
+
+      currentAsset.urls = this.getCopyUrlButtons(currentAsset.url, currentAsset.external_url);
```

Comment:
I'm assuming that each asset has a url and an external url. Is that reasonable? Conceptually, I don't see why it would ever be null. But if not, we should discuss what to do in the case that the attribute is null. Do we not render a button? If both are null, is the cell just empty?

Reply:
I think it's reasonable to assume each asset has both urls, then not render anything if that assumption fails and the attribute is null